### PR TITLE
BUGFIX: fix consul verify_ssl merging

### PR DESCRIPTION
**Pull request by taylorchu**
There are 2 issues with the tls code in 0.5.3-rc1:
1. typo in merging
2. one if statement that checks whether tls is enabled in tls config

### CODE REVIEW

**Comment by taylorchu** [comment]
@dadgar Is this ok?

File: nomad/structs/config/consul.go [hunk 1]
```diff
@@ -180,19 +180,21 @@ func (c *ConsulConfig) ApiConfig() (*consul.Config, error) {
...
-			},
+
+		if tlsConfig.InsecureSkipVerify {
+			config.HttpClient.Transport = &http.Transport{
+				TLSClientConfig: &tls.Config{
+					InsecureSkipVerify: true,
+				},
+			}
+		} else {
+			tlsClientCfg, err := consul.SetupTLSConfig(&tlsConfig)
```
Comment (dadgar) [inline nomad/structs/config/consul.go]: Can we just keep the stuff in the `else` block and delete the if/else. Why do we need to special case the `if tlsConfig.InsecureSkipVerify` case when `consul.SetupTLSConfig()` handles it?

**Comment by dadgar** [comment]
@taylorchu Good catch! Left one comment. Maybe I am missing something there

File: nomad/structs/config/consul.go (hunk 1 above)
Comment (taylorchu) [inline nomad/structs/config/consul.go]: yes. That else block is not needed after the consul code is updated. I even think that block creates some other bugs before but added anyway.

**Comment by dadgar** [comment]
Thanks!

**Pull request merged**